io-engine/src/grpc/mod.rs

```diff
@@ -114,18 +116,27 @@ impl GrpcClientContext {
     }
 }
 
-/// trait to lock serialize gRPC request outstanding
+/// Trait to lock serialize gRPC request outstanding.
 #[async_trait::async_trait]
 pub(crate) trait Serializer<F, T> {
     async fn locked(&self, ctx: GrpcClientContext, f: F) -> Result<T, Status>;
 }
 
+/// Trait allows Service implementing it to be locked by other Services along
+/// with usual serializing.
 #[async_trait::async_trait]
 pub(crate) trait RWSerializer<F, T> {
     async fn locked(&self, ctx: GrpcClientContext, f: F) -> Result<T, Status>;
     async fn shared(&self, ctx: GrpcClientContext, f: F) -> Result<T, Status>;
 }
 
+/// Trait allows service implementing to return read lock of itself to the
+/// caller.
+#[async_trait::async_trait]
+pub(crate) trait ReadLock {
```

Comment:
Suggest making it more generic, RWLock, otherwise you'd need to add yet a new trait or modify this one if we ever need to also write lock